docs/modules/engines/pages/clustering.adoc

```diff
@@ -0,0 +1,19 @@
+
+:imagesdir: ../assets/images
+= Clustering Engine
+
+The Clustering engine is a xref:introduction.adoc[correlation engine] that builds alarm clusters using the https://en.wikipedia.org/wiki/DBSCAN[DBSCAN algorithm].
+It draws information from the locally-persisted network inventory graph; this graph is automatically created and maintained by ALEC, and is based upon information from the associated xref:datasources:overview.adoc[datasource].
+
+== DBSCAN algorithm
+
+The DBSCAN algorithm that forms the basis for the Clustering engine measures the distance between alarms on the network inventory graph to determine whether they should be correlated into a cluster.
+The distance function calculates a weighted combination of distance in both time and space (on the shortest possible path between affected nodes) between alarms.
+
+== Performance
+
+The Clustering engine performs well when fewer than 500 candidate alarms exist.
```

Comment:
```suggestion
The clustering engine performs well when fewer than 500 candidate alarms exist.
```

Reply:
What happens when there are more than 500? Is it possible that users would ever have that many alarms? If they already have over 500 alarms when they install ALEC, what happens? Is it just super slow?